rotkehlchen/data_import/manager.py

```diff
@@ -0,0 +1,55 @@
+from pathlib import Path
+from typing import Optional, Tuple
+
+from rotkehlchen.data_import.importers.binance import BinanceImporter
+from rotkehlchen.data_import.importers.bisq_trades import BisqTradesImporter
+from rotkehlchen.data_import.importers.blockfi_trades import BlockfiTradesImporter
+from rotkehlchen.data_import.importers.blockfi_transactions import BlockfiTransactionsImporter
+from rotkehlchen.data_import.importers.cointracking import CointrackingImporter
+from rotkehlchen.data_import.importers.cryptocom import CryptocomImporter
+from rotkehlchen.data_import.importers.nexo import NexoImporter
+from rotkehlchen.data_import.importers.shapeshift_trades import ShapeshiftTradesImporter
+from rotkehlchen.data_import.importers.uphold_transactions import UpholdTransactionsImporter
+from rotkehlchen.data_import.utils import BaseImporter
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.types import DataSource
+
+
+class DataImporter:
+    def __init__(self, db: DBHandler):
+        self.db = db
+
+    def import_csv(
+            self,
+            source: DataSource,
+            filepath: Path,
+            timestamp_format: Optional[str],
+    ) -> Tuple[bool, str]:
+        importer: BaseImporter
+        if source == DataSource.COINTRACKING:
+            importer = CointrackingImporter(db=self.db)
+        elif source == DataSource.CRYPTOCOM:
+            importer = CryptocomImporter(db=self.db)
+        elif source == DataSource.BLOCKFI_TRANSACTIONS:
+            importer = BlockfiTransactionsImporter(db=self.db)
+        elif source == DataSource.BLOCKFI_TRADES:
+            importer = BlockfiTradesImporter(db=self.db)
+        elif source == DataSource.NEXO:
+            importer = NexoImporter(db=self.db)
+        elif source == DataSource.SHAPESHIFT_TRADES:
+            importer = ShapeshiftTradesImporter(db=self.db)
+        elif source == DataSource.UPHOLD_TRANSACTIONS:
+            importer = UpholdTransactionsImporter(db=self.db)
+        elif source == DataSource.BISQ_TRADES:
+            importer = BisqTradesImporter(db=self.db)
+        elif source == DataSource.BINANCE:
+            importer = BinanceImporter(db=self.db)
+        else:
+            # Need this for typing. Otherwise, importer might be referenced before assignment
+            raise AssertionError('Unsupported csv source')
+
+        success, msg = importer.import_csv(filepath=filepath, timestamp_format=timestamp_format)
+        importer.flush_trades()
+        importer.flush_asset_movements()
+        importer.flush_ledger_actions()
```

Comment:
Question: Isn't it counterintuitive to do `importer.flush_trades` and the likes after a CSV has been **imported** already?

I mean a success msg and boolean has been returned already. What do you think

Reply:
Hm... yes, I agree that it is not good. But needs thinking since I currently don't have an idea on how to change it, do you have?

Reply:
One thing that comes to my mind is...
- Move the `import_csv` method to `_import_csv`
```py
@abstractmethod
def _import_csv(self, filepath: Path, **kwargs: Any) -> Tuple[bool, str]:
    ...
```

- Now `import_csv` just calls `_import_csv` and calls other cleanup(this might not be the right word) e.g `flush_trades()` depending on the status of `_import_csv`.

Reply:
Okay, I put the cleanup into each importer before `return True, ''` lines. I'll push and then pls check if this is okay from your perspective